### ISSUE TITLE
Test that we build what we think we build

### ISSUE THREAD

**Issue by mpg** [Mbed-TLS/mbedtls#7430]
Some modules have various implementations depending on the target platform and toolchain, for example AES can use a portable C implementation, or AES-NI, AES-CE, Padlock. The logic for selecting the correct version for each platform and toolchain is not always trivial, so it should be tested.

Possible ways to do this could include grepping the object files for some function or instruction we're expecting to find or not to find, or having test programs print out which version is used and grepping the output.

**Comment by gilles-peskine-arm** [comment]
> grepping the object files for some function or instruction we're expecting to find or not to find

That can help, but for the example of AESNI/AESCE, we'd want to know that it's detected correctly at runtime, too.

> having test programs print out which version is used and grepping the output

Since 3.4.0/2.28.3, the `selftest` program indicates which AES implementation it uses.

**Comment by yuhaoth** [comment]
>Possible ways to do this could include grepping the object files for some function or instruction we're expecting to find or not to find. 

I do not think we should grep object files. I think we should use qemu to cover this case. [#7004](https://github.com/Mbed-TLS/mbedtls/pull/7004/files#diff-23f4aaf649ccf5bbb1adf53b50af6140b91e08f36077b5d3dae5e56bb0fcc593R79) mention same issue
